### PR TITLE
Add FLEX power constraint

### PR DESCRIPTION
**Pull request by qluo0320github**
## Description
This PR fixes the bug for FLEX resources. In addition to the maximum limit on "charge", the PR adds maximum on the "discharge" amount for FLEX based on the capacity and load shifting efficiency

## What type of PR is this? (check all applicable)

- [X] Bug Fix

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and .md files under /docs/src have been updated if necessary.
- [X] The latest changes on the target branch have been incorporated, so that any conflicts are taken care of before merging. This can be accomplished either by merging in the target branch (e.g. 'git merge develop') or by rebasing on top of the target branch (e.g. 'git rebase develop'). Please do not hesitate to reach out to the GenX development team if you need help with this.
- [x] Code has been tested to ensure all functionality works as intended.
- [x] CHANGELOG.md has been updated (if this is a 'notable' change).
- [x] I consent to the release of this PR's code under the GNU General Public license.

## How this can be tested
Tested with the WECC system

## Post-approval checklist for GenX core developers
After the PR is approved

- [x] Check that the latest changes on the target branch are incorporated, either via merge or rebase
- [x] Remember to squash and merge if incorporating into develop

### CODE REVIEW

File: src/model/resources/flexible_demand/flexible_demand.jl [hunk 1]
```diff
@@ -108,8 +108,10 @@ function flexible_demand!(EP::Model, inputs::Dict, setup::Dict)
                 [y in FLEX_Z, t = 1:T],
                 EP[:vCHARGE_FLEX][y, t] <= inputs["pP_Max"][y, t] * EP[:eTotalCap][y]
                 # NOTE: no maximum discharge rate unless constrained by other factors like transmission, etc.
-            end)
 
+                [y in FLEX_Z, t = 1:T],
+                flexible_demand_energy_eff(gen[y]) * EP[:vP][y, t] <= EP[:eTotalCap][y]
+            end)
         for y in FLEX_Z
```
[the remark this answers is not in the document]
Comment (lbonaldo) [reply src/model/resources/flexible_demand/flexible_demand.jl:115]: Yes, I should. Thanks @cfe316!

[the remark this answers is not in the document]
Comment (lbonaldo) [reply src/model/resources/flexible_demand/flexible_demand.jl:115]: actually, we can edit it

[the remark this answers is not in the document]
Comment (qluo0320github) [reply src/model/resources/flexible_demand/flexible_demand.jl]: Thanks! I just edited it.

[the remark this answers is not in the document]
Comment (lbonaldo) [reply src/model/resources/flexible_demand/flexible_demand.jl]: @qluo0320github could you also include the new constraint to the docstring?

[the remark this answers is not in the document]
Comment (qluo0320github) [reply src/model/resources/flexible_demand/flexible_demand.jl]: Thanks @lbonaldo. I just added it. I did this before but somehow it disappeared.

[the remark this answers is not in the document]
Comment (lbonaldo) [reply src/model/resources/flexible_demand/flexible_demand.jl]: Great, thank you @qluo0320github!

**Review by lbonaldo — APPROVED** [review]
Looks good.

**Pull request merged**